### DISCUSSION TITLE
How to set secret key on Mac

### DISCUSSION THREAD

**Discussion by cdaein** [go-shiori/shiori#999] [Q&A] [answered]
Hi,
I am running the Shiori server at startup, but I don't know how or where to set `SHIORI_HTTP_SECRET_KEY`. Without setting the key, it still works but I have to keep logging back in.

Here is how I set up my `.plist` inside `~/Library/LaunchAgents`:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
  <key>Label</key>
  <string>local.app.shiori</string>
  <key>ProgramArguments</key>
  <array>
    <string>/Users/[USERNAME]/shiori</string>
    <string>server</string>
    <string>--storage-directory</string>
    <string>/Users/[USERNAME]/shiori/</string>
    </array>
  <key>RunAtLoad</key>
  <true/>
</dict>
</plist>
```

I have set the `SHIORI_HTTP_SECRET_KEY` as environment variable inside `.zshrc` file, but it doesn't get loaded at startup time when the server starts. Any suggestions would be appreciated.

**Comment by Monirzadeh** [comment]
Can you set SHIORI_HTTP_SECRET_KEY with export and test if it work correctly?
```bash
export SHIORI_HTTP_SECRET_KEY=something
./shiori server
```

**Reply by cdaein** [reply]
Under normal circumstances, I think that would work, but because I'm using Mac's Launch Agent to start Shiori server at login time, environment variables are not yet read in.

If there is a way to set the secret key as CLI flag along with `shiori server` command, I think that would work, but I don't think it's supported.

**Comment by cdaein** [comment, marked as answer]
I found a way by using `EnvironmentVariables` key in the `.plist` file.

After the first login (and saving info) to Shiori in the browser, the login state persists even after computer restarts.

I think it would be helpful to add this to the Shiori doc for other Mac users.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
  <key>Label</key>
  <string>local.app.shiori</string>
  <key>EnvironmentVariables</key>
  <dict>
      <key>SHIORI_HTTP_SECRET_KEY</key>
      <string>somerandomvalue123489</string>
  </dict>
  <key>ProgramArguments</key>
  <array>
    <string>/Users/[USERNAME]/shiori</string>
    <string>server</string>
    <string>--storage-directory</string>
    <string>/Users/[USENAME]/shiori/</string>
    </array>
  <key>RunAtLoad</key>
  <true/>
  <key>ServiceDescription</key>
  <string>Shiori Bookmarking Service</string>
</dict>
</plist>
```

reference: https://launchd.info/

**Reply by Monirzadeh** [reply]
PR welcome.
Can you please add that to the documentation?

**Reply by cdaein** [reply]
Will do when I get some time later today.
